index.bs

```diff
@@ -345,6 +345,81 @@ and on when to use promises and when not to use promises,
 see <strong><a href="https://www.w3.org/2001/tag/doc/promises-guide">Writing
 Promise-Using Specifications</a></strong>.
 
+<h3 id="secure-context">Limit new features to secure contexts</h3>
+
+New capabilities added to the Web should be available only in
+<a href="https://w3c.github.io/webappsec-secure-contexts/">secure contexts</a>.
+Exposing them in non-secure contexts
+is discouraged and requires strong justification.
+The TAG is interested in hearing about and discussing cases
+where it is unclear whether exposing the capability
+in non-secure contexts is justifiable.
+
+This restriction exists for two reasons.
+First, it helps encourage Web content and applications
+to migrate to secure contexts.
+Second, some new features depend on authentication, integrity, or confidentiality
+to prevent substantial increases to the privacy or security risks of using the Web.
+For more detail, see the W3C TAG Finding on
+<a href="https://www.w3.org/2001/tag/doc/web-https">Securing the Web</a>.
+
+The most common justification that we expect
+for supporting a new capability in non-secure contexts
+is that the new capability is not recognizably separate
+from an existing feature that is available in non-secure contexts.
+In this case, limiting it to secure contexts could cause
+developer confusion about where the boundaries are.
+We also don't want to increase
+the complexity of implementations of Web technology
+by requiring tests for secure contexts in too many *types* of places.
+For example, exposing to non-secure contexts a new CSS property
+that adds support for a new line spacing model
+cannot be justified on this basis,
+but exposing the ability to omit the commas in the CSS ''rgb()'' function can.
+
+The existence of new features should generally be detectable,
+so that web content can act appropriately if the feature is present or not.
+This applies both to features that are not present because they are not implemented,
+and to features that are not present in a non-secure context because
+they are limited to secure contexts.
+Putting this another way,
+a feature that is limited to secure contexts should, in non-secure contexts,
+be indistinguishable from a feature that is not implemented.
+However, if, for some reason
+(a reason that itself requires serious justification),
+it is not possible for developers to detect whether a feature is present,
+limiting the feature to secure contexts
+might cause problems
+for libraries that may be used in either secure or non-secure contexts.
+
+If a feature depends on
+the expectations of authentication, integrity, or confidentiality
+that are met only in secure contexts,
+then it must be limited to secure contexts,
+even if the other factors above could justify exposing it in non-secure contexts.
```

Comment:
I don't see it as the feature depending on authentication, integrity, or confidentiality, but instead the feature posing some risk to user privacy or security which is mitigated only by requiring authentication, integrity, and confidentiality. I mean, at some level, _all_ features depend on the page's integrity, right? :)

WDYT about something like "If a feature poses a risk to user privacy or security which can be mitigated by requiring authentication, integrity, and confidentiality, the feature must be limited to secure contexts, ..."?

Reply:
To avoid the "poses a risk" in your wording that makes it sound like the feature is problematic, I think I'm going to try:
> If a feature would pose a risk to user privacy or security  without the authentication, integrity, or confidentiality that is present only in secure contexts, then the feature must be limited to secure contexts, ...